### PR TITLE
Add error boundary around routes in web UI

### PR DESCRIPTION
**Pull request by Gargron**
Use the same error template for crashes in the code, network errors when loading components, and unmatched routes.

Screen Shot 2022-10-22 at 20 24 06

### CODE REVIEW

File: app/javascript/mastodon/features/ui/util/react_router_helpers.js [hunk 1]
```diff
@@ -42,8 +42,38 @@ export class WrappedRoute extends React.Component {
...
+      hasError: true,
+    };
+  };
+
+  state = {
+    hasError: false,
+    stacktrace: '',
+  };
+
+  componentDidCatch (error, componentInfo) {
```
Comment (ykzts) [inline app/javascript/mastodon/features/ui/util/react_router_helpers.js:56]:
```suggestion
  componentDidCatch (error) {
```

File: app/javascript/mastodon/features/ui/util/react_router_helpers.js [hunk 2]
```diff
@@ -42,8 +42,38 @@ export class WrappedRoute extends React.Component {
     componentParams: {},
   };
 
+  static getDerivedStateFromError (error) {
```
Comment (ykzts) [inline app/javascript/mastodon/features/ui/util/react_router_helpers.js:45]:
```suggestion
  static getDerivedStateFromError () {
```

**Review by ykzts — CHANGES_REQUESTED** [review]

**Review by ykzts — APPROVED** [review]

File: app/javascript/mastodon/features/ui/components/bundle_column_error.js [hunk 3]
```diff
@@ -1,44 +1,155 @@
...
+    const { onRetry } = this.props;
+
+    if (onRetry) {
+      onRetry();
+    }
   }
 
   render () {
-    const { multiColumn, intl: { formatMessage } } = this.props;
+    const { errorType, multiColumn, stacktrace, intl } = this.props;
```
Comment (ykzts) [inline app/javascript/mastodon/features/ui/components/bundle_column_error.js:119]:
```suggestion
    const { errorType, multiColumn, stacktrace } = this.props;
```

**Review by ykzts — CHANGES_REQUESTED** [review]

**Review by ykzts — APPROVED** [review]

**Comment by ClearlyClaire** [comment]
This could use a link to the issue tracker…

**Comment by ClearlyClaire** [comment]
Additionally, when the column is pinned in multi-column view, it could offer to unpin the column.

**Pull request merged**